### ISSUE TITLE
Example SharingWithUnet still relevant?

### ISSUE THREAD

**Issue by NAMiedema** [Microsoft/MixedRealityToolkit-Unity#2102]
## Overview
The sharing with UNET example does not seem to work in our tests, we've tried multiple setups trying to run on exclusively on HoloLenses (2 lenses), using a server in the editor and HoloLenses as clients, and using a HoloLens as host and the editor as a client. In none of the tests the network discovery worked, moreover testing to host a simple test application on the HoloLens did not work. Is the example still relevant? We're currently not sure how to use anchors and share them, if someone has a functioning example using UNET that would be a great help too.

## Expected Behavior
Network discovery shows all hosts on the lan.

## Actual Behavior
No HoloLenses are found?

## Steps to reproduce
Running the SharingWithUnet example scene.

## Unity Editor Version
2017.2.0f3

## Mixed Reality Toolkit Release Version
 2017.2.1.4

**Comment by FejZa** [comment]
Last time I checked it was working fine. I am working on some updates to the UNET components I'd like to add to the vNext version of the Toolkit. In the meantime you could check out the SpectatorView sample project "SharedHolograms" here:

https://github.com/Microsoft/MixedRealityCompanionKit/tree/master/SpectatorView/Samples

It uses UNET for sharing the anchor. Of course it contains some stuff you will not need if you don't have a SpectatorView. But generally the sample should do what you need. I will check the UNET example in the MRTK too.

When using UNET you need to make sure all devices are properly connected to the network. Also make sure it's a private network and not a public network.

**Comment by NAMiedema** [comment]
Thanks for the quick reply, we'll try the example later today. We were using a local network, we'll double check whether the network is private or not.

**Comment by NAMiedema** [comment]
We're using a local router connected to the internet, all HoloLenses and editors are connected to this network, does the HoloLens see this network as a private one? On windows desktop you always get the choice to set a network to private but this seems to be absent on the HoloLens.

**Comment by FejZa** [comment]
Yes, you're right. On HoloLens you don't get to chose. Sorry, I was thinking about when hosting using the editor. Have you tried hosting a session in the Unity editor and then checking if the HoloLens can discover it?

**Comment by NAMiedema** [comment]
Yes we have hosted in the Editor, and the lenses can connect in this case, also via network discovery, the issue in this case is that no anchors are set in the game. Looking at the code it seems that the example was intended for hosting on the HoloLens which we're still not able to do.

**Comment by FejZa** [comment]
Network Discovery should work even without anchors. The anchors is just for syncing content position. The example was working fine for me. Did you have any success so far? Try e.g. disabling the Windows Firewall and then hosting a session using the editor. Then check if the HoloLens can see it. Also you can try and create a Hotspot using your mobile device. Worked for me too. There must be some issue with the network configuration.

**Comment by keveleigh** [comment]
The SharingWIthUNET scene uses [UNetDiscovery](https://docs.unity3d.com/Manual/UNetDiscovery.html), which requires UDP Broadcast to be enabled on your router/network. You can likely turn this on in your router settings.

![image](https://user-images.githubusercontent.com/3580640/41626157-367ed0ec-73d0-11e8-89bb-fad0943292b5.png)

This seems to be the most common issue if devices can't see each other.

I just built and ran the SharingWithUnetExample and was able to get two HoloLens devices in the session, with one of the HoloLens devices hosting. I didn't make any changes to the scene.
